src/writer/tests.rs

```diff
@@ -61,7 +61,11 @@ proptest! {
         }
 
         let capacity_pages = memory.size();
-        let min_pages_required = (offset + bytes.len() as u64 + WASM_PAGE_SIZE - 1) / WASM_PAGE_SIZE;
+        let min_pages_required = if bytes.is_empty() {
+            0
+        }else{
+            (offset + bytes.len() as u64 + WASM_PAGE_SIZE - 1) / WASM_PAGE_SIZE
+        };
```

Comment:
nit: I think we can use div_ceil here for clarity.

```suggestion
        } else {
            u64::div_ceil(offset + bytes.len() as u64, WASM_PAGE_SIZE)
        };
```